src/Neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -272,7 +275,8 @@ private ContractTask Update(ApplicationEngine engine, byte[] nefFile, byte[] man
 
             engine.AddGas(engine.StoragePrice * ((nefFile?.Length ?? 0) + (manifest?.Length ?? 0)));
 
-            var contract = engine.Snapshot.GetAndChange(CreateStorageKey(Prefix_Contract).Add(engine.CallingScriptHash))?.GetInteroperable<ContractState>();
+            var contract = engine.Snapshot.GetAndChange(CreateStorageKey(Prefix_Contract).Add(engine.CallingScriptHash))?
+                .GetInteroperable<ContractState>(ExecutionEngineLimits.Default with { MaxStackSize = 4096 });
```

Comment:
Frankly, I'd use the "normal" limit and let the update transaction fail. It should not affect any balances (that contract was never called again (it couldn't be) and tx fee is the same for failed and successful cases).

Reply:
Totally Agree